test/integration/setup.js

```diff
@@ -32,17 +34,19 @@ before(() => {
   // set global variables
   global.chai = chai;
   global.expect = chai.expect;
-  global.agent = chai.request.agent(baseUrl);
+  global.agent = chai.request.agent(server);
   const { agent } = global;
 
   // base user defined in test data
   const user = { username : 'superuser', password : 'superuser', project : 1 };
 
   // trigger login
-  return (() => agent.post('/auth/login').send(user))();
+  return agent.post('/auth/login').send(user);
 });
 
 // runs after all tests are completed
-after(() => {
+after(done => {
+  delete require.cache[require.resolve('../../bin/server/app')];
```

Comment:
I'm not sure I understand why you need to do this.  It also looks really hacky.  Can we remove it?

Reply:
You're totally right, that's the reason why i talked about that